streams/src/test/java/org/apache/kafka/streams/kstream/SuppressedTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream;
+
+import org.apache.kafka.streams.kstream.Suppressed.BufferConfig;
+import org.junit.Test;
+
+import java.time.Duration;
+
+import static org.apache.kafka.streams.kstream.Suppressed.BufferConfig.boundedByKeys;
+import static org.apache.kafka.streams.kstream.Suppressed.BufferConfig.boundedBySize;
+import static org.apache.kafka.streams.kstream.Suppressed.BufferConfig.unbounded;
+import static org.apache.kafka.streams.kstream.Suppressed.IntermediateSuppression.withBufferConfig;
+import static org.apache.kafka.streams.kstream.Suppressed.IntermediateSuppression.withEmitAfter;
+import static org.apache.kafka.streams.kstream.Suppressed.emitFinalResultsOnly;
+import static org.apache.kafka.streams.kstream.Suppressed.intermediateEvents;
+
+public class SuppressedTest {
+
+    @Test
+    public void intermediateEventsShouldAcceptAnyBuffer() {
+        intermediateEvents(withEmitAfter(Duration.ofMillis(2)));
+        intermediateEvents(withBufferConfig(unbounded()));
+        intermediateEvents(withBufferConfig(boundedByKeys(2)));
+        intermediateEvents(withBufferConfig(boundedBySize(2L)));
+        intermediateEvents(withEmitAfter(Duration.ofMillis(2)).bufferConfig(unbounded()));
+        intermediateEvents(withBufferConfig(unbounded()).emitAfter(Duration.ofMillis(2)));
+        intermediateEvents(withEmitAfter(Duration.ofMillis(2)).bufferConfig(boundedByKeys(2)));
+        intermediateEvents(withBufferConfig(boundedByKeys(2)).emitAfter(Duration.ofMillis(2)));
+    }
+
+    @Test
+    public void finalEventsShouldAcceptOnlyStrictBuffers() {
```

Comment:
"shouldAccept ONLY" -- sound like a negative test but I don't see any exception?

The `doesn't compile because the buffer is eager` part is weird... The condition you express is ensured via the interface -- ie, it does not compile. Thus, I don't think we need a negative test.

Reply:
The purpose of this "test" is explicitly to demonstrate what compiles and what doesn't.

It's of course not possible to write code that doesn't compile in order to demonstrate that it doesn't compile, so I wrote the code and commented it out. If you'd like to "run" the test, you can uncomment the "negative test" lines and verify they are not permitted.

If you think this is silly, I can rename the test.

Reply:
Well, I guess nobody will ever uncomment those lines to test if it does not compile -- seems to be dead code to me.